common/src/main/scala/workers/WebWorkerF.scala

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package workers
+
+import cats.effect.Async
+import cats.effect.Sync
+import cats.effect.std.Dispatcher
+import fs2.Stream
+import fs2.concurrent.Channel
+import org.scalajs.dom
+
+import scala.scalajs.js
+
+/**
+ * WebWorker abstraction running on F. it is possible to post messages and get a stream of events
+ * from the web worker
+ */
+trait WebWorkerF[F[_]] {
+
+  /**
+   * Post a message on effect F
+   */
+  def postMessage(message: js.Any): F[Unit]
+
+  /**
+   * Terminate the web worker
+   */
+  def terminate: F[Unit]
+
+  /**
+   * Streams of events from the worker
+   */
+  def stream: Stream[F, dom.MessageEvent]
+}
+
+object WebWorkerF {
+
+  def apply[F[_]: Async](worker: dom.Worker) = new WebWorkerF[F] {
```

Comment:
Can we make this return a `Resource[WebWorker[F]]` that calls `terminate` when closing? That way clients are freed from the onus of manually terminating.

This won't make much of a difference if we use global web workers, but if we ever use component-local ones we can easily use them with the `useResource` hook.

Reply:
Good idea